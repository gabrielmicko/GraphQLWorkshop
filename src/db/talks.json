[
  {
    "id": "1",
    "date": "Aug 3 - 10:30",
    "duration": 50,
    "talkType": "keynote",
    "talkTitle": "JavaScript: clouds on the horizon",
    "speakerId": "1"
  },
  {
    "id": "2",
    "date": "Aug 3 - 11:20",
    "duration": 40,
    "talkType": "talk",
    "talkTitle": "Custom RxJS operators",
    "speakerId": "2"
  },
  {
    "id": "3",
    "date": "Aug 3 - 12:30",
    "duration": 40,
    "talkType": "talk",
    "talkTitle": "From code to pixels in a Blink",
    "speakerId": "3"
  },
  {
    "id": "4",
    "date": "Aug 3 - 13:10",
    "duration": 50,
    "talkType": "talk",
    "talkTitle": "Vue + GraphQL",
    "speakerId": "4"
  },
  {
    "id": "5",
    "date": "Aug 3 - 15:20",
    "duration": 10,
    "talkType": "lightning",
    "talkTitle": "React with Blackjack and Hooks",
    "speakerId": "5"
  },
  {
    "id": "6",
    "date": "Aug 3 - 15:30",
    "duration": 10,
    "talkType": "lightning",
    "talkTitle": "Accessibility through the eyes of users",
    "speakerId": "6"
  },
  {
    "id": "7",
    "date": "Aug 3 - 15:40",
    "duration": 10,
    "talkType": "lightning",
    "talkTitle": "Tensorflow in Action",
    "speakerId": "7"
  },
  {
    "id": "8",
    "date": "Aug 3 - 15:50",
    "duration": 10,
    "talkType": "lightning",
    "talkTitle": "Language Independent Validation Rules 2.0",
    "speakerId": "8"
  },
  {
    "id": "9",
    "date": "Aug 3 - 16:00",
    "duration": 10,
    "talkType": "lightning",
    "talkTitle": "Reactive programming. Svelte way",
    "speakerId": "9"
  },
  {
    "id": "10",
    "date": "Aug 3 - 16:10",
    "duration": 10,
    "talkType": "lightning",
    "talkTitle": "ML + AR in js (to be or not to be)",
    "speakerId": "10"
  },
  {
    "id": "11",
    "date": "Aug 3 - 16:50",
    "duration": 40,
    "talkType": "talk",
    "talkTitle": "Fast future with WebAssembly",
    "speakerId": "11"
  },
  {
    "id": "12",
    "date": "Aug 3 - 17:30",
    "duration": 40,
    "talkType": "keynote",
    "talkTitle": "A universal fusion of the team",
    "speakerId": "12"
  },
  {
    "id": "13",
    "date": "Aug 4 - 11:00",
    "duration": 150,
    "talkType": "workshop",
    "talkTitle": "Building GraphQL Server with JS",
    "speakerId": "13"
  },
  {
    "id": "14",
    "date": "Aug 4 - 14: 30",
    "duration": 90,
    "talkType": "workshop",
    "talkTitle": "React with Blackjack and Hooks",
    "speakerId": "5"
  },
  {
    "id": "15",
    "date": "Aug 4 - 17:00",
    "duration": 90,
    "talkType": "workshop",
    "talkTitle": "Reactive programming. Svelte way",
    "speakerId": "9"
  }
]
