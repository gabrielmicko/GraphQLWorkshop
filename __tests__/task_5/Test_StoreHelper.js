import speakers from '../../src/db/speakers.json';
import talks from '../../src/db/talks.json';
import {
  setTestDefaults,
  getDataStore,
  insertData,
  wipeTable,
  getAll,
  getByFilters,
  removeFile,
} from '../../src/store/helper';

describe('tests task 5', () => {
  beforeEach(() => {
    setTestDefaults(speakers, talks);
  });

  it('getByFilters should return filtered data set', async () => {
    const selectedSpeaker = getByFilters('speakers', {
      id: '13',
    });
    expect(selectedSpeaker).toMatchSnapshot();
  });

  it('getAll should return all data set', async () => {
    const everySpeaker = getAll('speakers');
    expect(everySpeaker).toMatchSnapshot();
  });

  it('wipeTable should remove everything', async () => {
    wipeTable('speakers');
    const everySpeaker = getAll('speakers');

    expect(everySpeaker).toStrictEqual([]);
  });

  it('insertData should insert a record', async () => {
    const exampleData = {
      id: 'X',
      test: 'test',
    };
    insertData('speakers', exampleData);
    const everySpeaker = await getByFilters('speakers', exampleData);
    expect(everySpeaker).toStrictEqual([exampleData]);
  });
});
