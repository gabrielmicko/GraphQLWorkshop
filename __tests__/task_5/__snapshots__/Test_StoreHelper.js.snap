// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests task 5 getAll should return all data set 1`] = `
Array [
  Object {
    "bio": "15 years of JS everywhere: from microcontrollers to rendering video in the cloud. More than 6 years of educational experience (at two universities and Illya’s own courses), Ph.D. in Computer Science (field of interest: System Analysis and Theory of optimal decisions). Now works as front-end developer in GitLab.",
    "id": "1",
    "name": "Illya Klymov",
    "twitter": "https://twitter.com/xanf_ua",
  },
  Object {
    "bio": "Chris is a senior frontend developer at Scalac. He’s been working with Angular since the first beta came out and he loved it. Chris is passionate about functional programming in JavaScript and loves working with RxJS. In his free time he likes to go jogging.",
    "github": "https://github.com/ktrz",
    "id": "2",
    "name": "Chris Trześniewski",
    "twitter": "https://twitter.com/ktrz__",
  },
  Object {
    "bio": "Prashant is a frontend developer at Frontmen in the Netherlands. Apart from coding, he is also responsible to create a technical trajectory for developers within Frontmen. While not coding, he likes to travel, preach about technologies and break a sweat at the basketball court with friends.",
    "github": "https://www.linkedin.com/in/prashantpalikhe/",
    "id": "3",
    "linkedin": "https://github.com/prashantpalikhe/",
    "name": "Prashant Palikhe",
    "twitter": "https://twitter.com/PrashantPalikhe",
  },
  Object {
    "bio": "Natalia Tepluhina is a big fan of Vue.js framework and contributor to Vue ecosystem. Currently she’s working as a part of Vue.js docs team. Natalia is a constant learner and loves to share knowledge with other people. She is a conference speaker and author of articles on different topics related to Vue.js . Thanks to this activities she has got a title of Google Developer Expert in Web Technologies.",
    "github": "https://gitlab.com/ntepluhina",
    "id": "4",
    "name": "Natalia Tepluhina",
    "twitter": "https://twitter.com/N_Tepluhina",
  },
  Object {
    "bio": "Experienced and passionate JavaScript/React developer. Speaker, lecturer, mentor, open source contributor, active member of Vinnytsia IT community.",
    "github": "https://github.com/yevhene",
    "id": "5",
    "linkedin": "https://www.linkedin.com/in/yevhene/",
    "name": "Yevhen Shemet",
  },
  Object {
    "bio": "Frontend developer accessibility consultant with experience in backend. Now I’m working at big projects for UK, where main magic criteria are accessibility and performance. I always wanted to do more than coding. And now the laws of many countries give me the opportunity to do it.",
    "id": "6",
    "linkedin": "https://www.linkedin.com/in/andriis",
    "name": "Andrii Shtadler",
  },
  Object {
    "bio": "A person who always tries something new and never gives up if it goes wrong. Coding is my passion and it is so easy for me to lose track of time.",
    "github": "https://github.com/Croftyland",
    "id": "7",
    "name": "Khrystyna Landvytovych",
    "twitter": "https://twitter.com/croftyland",
  },
  Object {
    "bio": "My name is Viktor Turskyi. I am CEO and principal architect at WebbyLab. Open source developer. More than 15 years in IT, launched more than 60 javascript projects, more than 30 talks at conferences",
    "github": "https://github.com/koorchik/my-talks",
    "id": "8",
    "linkedin": "https://www.linkedin.com/in/turskyi/",
    "name": "Viktor Turskyi",
    "twitter": "https://twitter.com/koorchik",
  },
  Object {
    "bio": "Accessibility advocate during free time and Green Bay Packers fan during cold nights.",
    "id": "9",
    "name": "Oleksandr Kovtun",
  },
  Object {
    "bio": "Senior full-stack javascript developer in company IG «RIA». I have experience in developing PWA (Progressive Web Apps), AMP (Accelerated Mobile Pages) and universal apps, build VR UI. I’m working on creating VR interfaces. I am interested in machine learning and AI.",
    "github": "https://boryamogila.github.io/",
    "id": "10",
    "linkedin": "https://www.linkedin.com/in/--06b6b212a",
    "name": "Borys Mohyla",
    "twitter": "https://twitter.com/BoryaMogila",
  },
  Object {
    "bio": "Milica is web developer from Belgrade, currently writing for Google about performance optimizations and tools that can help you with that. Before that she was an intern at Mozilla where she worked on Servo engine and learned a lot about how browsers work. She’s also a Mozilla Tech Speaker and a big believer in open web, always exploring new technologies that are moving it forward, like WebAssembly and WebVR.",
    "github": "https://github.com/mihajlija",
    "id": "11",
    "linkedin": "https://www.linkedin.com/in/milica-mihajlija-a0523014b/",
    "name": "Milica Mihajlija",
    "twitter": "https://twitter.com/bibydigital",
  },
  Object {
    "bio": "Paul is a web developer, who happens to be several communities organizer, public speaker, mentor, traveler and just a guy with a good sense of humor. Currently lives in Minsk, but you can easily find him at any international conference.",
    "id": "12",
    "name": "Paul Yuhnovich",
    "twitter": "https://twitter.com/Paul_Yuhnovich",
  },
  Object {
    "bio": "I am enthusiastic front-end developer who loves the web and new technologies around it. I am tech speaker, contributor, volunteer at Mozilla. I believe in the mission to ensure the Internet is a global public resource, open and accessible to all.",
    "github": "https://github.com/gabrielmicko",
    "id": "13",
    "name": "Gabriel Mičko",
    "twitter": "https://twitter.com/gabriel_micko",
  },
]
`;

exports[`tests task 5 getByFilters should return filtered data set 1`] = `
Array [
  Object {
    "bio": "I am enthusiastic front-end developer who loves the web and new technologies around it. I am tech speaker, contributor, volunteer at Mozilla. I believe in the mission to ensure the Internet is a global public resource, open and accessible to all.",
    "github": "https://github.com/gabrielmicko",
    "id": "13",
    "name": "Gabriel Mičko",
    "twitter": "https://twitter.com/gabriel_micko",
  },
]
`;
